dev/EVAndPATH/Microsoft.Process.Environment/module.g.cpp

```diff
@@ -0,0 +1,61 @@
+// WARNING: Please don't edit this file. It was generated by C++/WinRT v2.0.210122.3
+
+#include "pch.h"
+#include "winrt/base.h"
+void* winrt_make_Microsoft_Process_Environment_EnvironmentManager();
+
+bool __stdcall winrt_can_unload_now() noexcept
+{
+    if (winrt::get_module_lock())
+    {
+        return false;
+    }
+
+    winrt::clear_factory_cache();
+    return true;
+}
+
+void* __stdcall winrt_get_activation_factory([[maybe_unused]] std::wstring_view const& name)
+{
+    auto requal = [](std::wstring_view const& left, std::wstring_view const& right) noexcept
+    {
+        return std::equal(left.rbegin(), left.rend(), right.rbegin(), right.rend());
+    };
```

Comment:
[External]
Is this right? It's my understanding WinRT type names are not case-sensitive and you could get called asking to activate an instance of `microSOFT.process.environment.environmentMANAGER`.

Reply:
@riverar you're right.

@kennykerr  can you advise? This generated `*.g.cpp` file should really be `#included` into a module main of some kind.#385

Reply:
The intent of that documentation is to leave room for case-insensitive languages to call WinRT APIs without ambiguity. It is not to imply that [RoGetActivationFactory](https://docs.microsoft.com/en-us/windows/win32/api/roapi/nf-roapi-rogetactivationfactory) is case-insensitive - it is not and has never been.